piptools/scripts/compile.py

```diff
@@ -344,14 +355,50 @@ def cli(
             )
             raise click.BadParameter(msg)
 
-        if src_file == "-":
-            # pip requires filenames and not files. Since we want to support
-            # piping from stdin, we need to briefly save the input from stdin
-            # to a temporary file and have pip read that.  also used for
+        if src_file == "-" or (
+            os.path.basename(src_file).endswith(".py") and not is_setup_file
```

Comment:
pathlib is usually nicer
```suggestion
            Path(src_file).suffix == ".py" and not is_setup_file
```

Reply:
The src path should probably be a local variable (not temporary like in the suggestion), to be used for suffix check here and open / read_text on lines below.